code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/code_generation/DockerFileCodeGenerator.xtend

```diff
@@ -0,0 +1,157 @@
+                + "container exceeds the maximum of one service per container ")
+        }
+
+        /*
+         * The creation of the dockerfile is supported in two different ways.
+         * First we check is the dockerfile is explicitly defined in an aspect.
+         * Otherwise the dockerfile is build based on the name of the deployed microservice,
+         * operation environment and exposed port, defined by the container endpoints.
+         */
+        val dockerFileAspect = container.aspects.findFirst[aspect |
+            aspect.name.toLowerCase == "dockerfile"
+        ]
+
+        if (dockerFileAspect !== null) {
+            val dockerFile = new DockerFile(dockerFileAspect.propertyValues.get(0).value)
+
+            content.put(targetFolder
+                + File.separator
+                + Util.buildPathFromQualifiedName(container.deployedServices.get(0).qualifiedName)
+                + File.separator
+                + "Dockerfile",
+                dockerFile.toString
+            )
+        } else {
+            var dockerFile = new DockerFile
+            dockerFile.deployedServiceName = container.name.toLowerCase
+
+            if (container.deployedServices.get(0).version === null) {
+                dockerFile.version = "v0"
+            } else {
+                 dockerFile.version = container.deployedServices.get(0).version
+            }
+
+            dockerFile.version = container.deployedServices.get(0).version
+            dockerFile.operationEnvironment = container.operationEnvironment.environmentName
+
+            content.put(targetFolder
+                + File.separator
+                + Util.buildPathFromQualifiedName(
+                    container.deployedServices.get(0).qualifiedName.toLowerCase
+                )
+                + File.separator
+                + "Dockerfile",
+                dockerFile.toString
+            )
+        }
+    }
+
+    /*
+     * This method is responsible for generating the dockerfile. The file is generated depending on
+     * whether the infrastructure node has a DockerFileAspect or not.
+     */
+    private def createDockerFileForInfrastructureNode(IntermediateInfrastructureNode node) {
+           val dockerFileAspect = node.aspects.findFirst[aspect |
+               aspect.name.toLowerCase == "dockerfile"]
+
+           if (dockerFileAspect !== null) {
+                val dockerFile = new DockerFile(dockerFileAspect.propertyValues.get(0).value)
+
+                content.put(targetFolder
+                    + File.separator
+                    + node.name.toLowerCase
+                    + File.separator
+                    + "Dockerfile",
+                    dockerFile.toString
+                )
+            } else {
+                var dockerFile = new DockerFile
+                dockerFile.deployedServiceName = node.name.toLowerCase
+                dockerFile.version = "0.0.1-SNAPSHOT"
```

Comment:
Why is the dockerFile version `0.0.1-SNAPSHOT`? And: Make this a constant to enable quick adaptation in the future if necessary.